python/ray/data/_internal/logical/rules/add_streaming_repartition_when_zip.py

```diff
@@ -0,0 +1,80 @@
+import copy
+from typing import List
+
+from ray.data._internal.logical.interfaces import (
+    LogicalOperator,
+    LogicalPlan,
+    Rule,
+)
+from ray.data._internal.logical.operators.map_operator import StreamingRepartition
+from ray.data._internal.logical.operators.n_ary_operator import Zip
+
+
+class AddStreamingRepartitionWhenZip(Rule):
```

Comment:
You actually don't need the rule you can just wrap input ops into StreamingRepartition inside `ds.zip` method